src/averageAges.js

```diff
@@ -15,13 +15,28 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter(person => century === Math.ceil(person.died / 100)
+      && person.sex === 'm')
+    : people.filter(person => person.sex === 'm');
+
+  const ageOfDeath = men.map(man => man.died - man.born);
+
+  const sumofAge = ageOfDeath.reduce((sum, age) => sum + age, 0);
+
+  const averageOfMen = Number((sumofAge / ageOfDeath.length).toFixed(2));
+
+  return averageOfMen;
```

Comment:
this logic repeats in methods - create a helper function and use it in your methods instead of repeating each time